stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/AttachmentViewHolderFactory.kt

```diff
@@ -0,0 +1,87 @@
+package io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment
+
+import android.view.ViewGroup
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.ui.common.extensions.internal.isMedia
+import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
+import io.getstream.chat.android.ui.databinding.StreamUiItemFileAttachmentGroupBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemImageAttachmentBinding
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.AttachmentClickListener
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.AttachmentDownloadClickListener
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.AttachmentLongClickListener
+
+/**
+ * Factory for creating the attachment contents displayed within message items.
+ *
+ * Displays media and file attachments by default.
+ */
+public open class AttachmentViewHolderFactory(
+    private val attachmentClickListener: AttachmentClickListener,
+    private val attachmentLongClickListener: AttachmentLongClickListener,
+    private val attachmentDownloadClickListener: AttachmentDownloadClickListener,
+) {
+
+    public companion object {
+        private const val BUILT_IN_TYPES = 9000
+
+        /**
+         * The media attachment type.
+         */
+        public const val MEDIA: Int = BUILT_IN_TYPES + 1
+
+        /**
+         * The file attachment type.
+         */
+        public const val FILE: Int = BUILT_IN_TYPES + 2
+    }
+
+    /**
+     * Determine the view type to be used for the given list of attachments.
+     *
+     * Supports media and file attachments by default. Make sure to call into the
+     * super implementation when overriding this method if you want these default
+     * attachment types to be handled.
+     */
+    public open fun getItemViewType(attachments: List<Attachment>): Int {
+        // TODO update the logic for rendering attachments as files - match Compose implementation?
```

Comment:
Might want to do this within this PR too, but it's a behaviour change from what we have currently